columnphysics/icepack_shortwave.F90

```diff
@@ -295,29 +287,24 @@ subroutine shortwave_ccsm3 (aicen,    vicen,    &
                                fswsfc=fswsfc(n),     &
                                fswint=fswint(n),     &
                                fswthru=fswthru(n),   &
-                               fswthru_vdr=l_fswthru_vdr(n),&
-                               fswthru_vdf=l_fswthru_vdf(n),&
-                               fswthru_idr=l_fswthru_idr(n),&
-                               fswthru_idf=l_fswthru_idf(n),&
+                               fswthru_vdr=l_fswthru_vdr,&
+                               fswthru_vdf=l_fswthru_vdf,&
+                               fswthru_idr=l_fswthru_idr,&
+                               fswthru_idf=l_fswthru_idf,&
                                fswpenl=fswpenl(:,n), &
                                Iswabs=Iswabs(:,n))
 
          if (icepack_warnings_aborted(subname)) return
 
+         if(present(fswthru_vdr)) fswthru_vdr(n) = l_fswthru_vdr
+         if(present(fswthru_vdf)) fswthru_vdf(n) = l_fswthru_vdf
+         if(present(fswthru_idr)) fswthru_idr(n) = l_fswthru_idr
+         if(present(fswthru_idf)) fswthru_idf(n) = l_fswthru_idf
```

Comment:
This is a very nice cleanup because these statements are inside the condition on `aicen`. So this means that, if for some category the condition is not matched, whatever values were in `fswthru_idf` and friends stay there untouched. This was not the case before, since the whole arrays were overwritten by their local counterparts, which themselves were uninitialized. This lead to a crash in one of our configurations (when compiling with NaN initialization) which lead me to carry this commit in our internal fork: https://github.com/phil-blain/Icepack/commit/c4776379d95501d90ae79fd86d234472cd02d4f9

Nice to know I'll be able to drop this one eventually :)